### ISSUE TITLE
Two apps are skipped when pressing `<tab>` once, occasionally

### ISSUE THREAD

**Issue by itsamoreh** [lwouis/alt-tab-macos#1989]
**Describe the bug**

While pressing tab to navigate through the window list, sometimes two apps are skipped over.

This makes it hard to select the window I want to switch to because I'll accidentally end up selecting the next one in the list because it rapidly skips past two apps even though I only pressed `<tab>` once.

**Screenshots / video**

There is sound in the screen recording so you can hear keypresses in addition to them being displayed on screen.

You can see the issue when the alt-tab window rapidly skips past two apps even though I only pressed `<tab>` once. The whole time I'm pressing tab at a constant, slow rate so this rapid skipping is unexpected. It seems to happen randomly.

https://user-images.githubusercontent.com/12365909/193635162-9aa9ac95-4024-4cc0-827a-db87ac0d89de.mov

**Steps to reproduce the bug**

1. Open a bunch of windows
2. Use the `command + <tab>` shortcut to navigate through windows in the list
3. Press `<tab>` multiple times while holding `command` to navigate through the apps
4. Observe that occasionally, two apps are skipped over even though you only pressed `<tab>` once

**Comment by itsamoreh** [comment]
After watching the video a couple of times, it seems like this might be caused by a lag and I am actually pressing `<tab>` twice. But after the first press there is a lag then it skips two quickly after pressing `<tab>` again.

It's really hard to tell though. Either way still seems like a bug to me.

**Comment by XInTheDark** [comment]
It is definitely not a bug - such erratic behaviour would not normally exist, and it would have been noticed by many people if it occurred. Also, in the video, it seems that you pressed the `tab` key twice in a row; it was not the app's fault.

**Comment by lwouis** [comment]
> The whole time I'm pressing tab at a constant, slow rate

You're pressing pretty quickly in my opinion. So much so that it's hard to follow what's happening on the video. Could you please tell us the exact timing when you think the issue is exhibited?

I think AltTab is working as intended, most likely.

* Are you using an external keyboard, and/or software that remaps/interacts with the keys?
* What's your key repeat rate? AltTab complies to what you set in `System Preferences` > `Keyboard` > Key Repeat` and `Delay Until Repeat`, so if you have unusual values here, it could be involved in the issue.

**Comment by itsamoreh** [comment]
@XInTheDark @lwouis 

I'm sorry that my original video wasn't clear.

Here is another quick video where you can see and hear the issue happen three times in a row. The issue should be much more obvious in this video. The bug happens at 12, 15, and 19 seconds in the video when an app in the list is rapidly skipped.

You can see that each time (other than the time I skip multiple apps) I'm only pressing tab twice as reported by keycastr. I also took this screen recording with sound so you can also hear me press tab just twice. Yet three apps are skipped three times in a row. You can see the correct behaviour happen before and after the buggy behaviour.

This bug makes alt-tab very unreliable for me and I can assure you it is real. If it's not clear in this video let me know and I will try to take an even better screen recording.

I've attached a screenshot of my "Key Repeat" and "Delay Until Repeat" settings. I actually reduced my key repeat setting before taking this screen recording but it didn't fix the problem. I am using an external keyboard but I don't experience this problem in any other program.

https://user-images.githubusercontent.com/12365909/196787914-56c2b6f9-6ba4-4d3d-adc0-791d02214238.mov

<img width="780" alt="screen-capture 2022-10-19 at 3 41 03 PM" src="https://user-images.githubusercontent.com/12365909/196788094-872cc3cd-e387-49cf-b5c9-182145fd48b7.png">

**Comment by lwouis** [comment]
> I am using an external keyboard

If you're using a laptop, can you reproduce on the built-in/internal keyboard? It would be nice to really make sure the keyboard is not the cause here, since other users haven't reported this issue.

**Comment by itsamoreh** [comment]
I can reproduce this with the built-in keyboard on my 2021 16" M1 MacBook Pro.

When I hold `tab`, it goes through the apps much slower than when the bug happens.

**Comment by lwouis** [comment]
I can't reproduce it with the steps in the OP so i'm not sure what to do. Any conditions that make it happen more often?

**Comment by itsamoreh** [comment]
I upgraded to MacOS Ventura a couple of days ago and this is no longer happening. I'll follow up if it happens again.

**Comment by lwouis** [comment]
Good news. I'll close the issue. If it still happens let me know and i'll re-open it
